flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/FlinkDistribution.java

```diff
@@ -254,22 +259,33 @@ public void submitSQLJob(SQLJobSubmission job) throws IOException {
 		AutoClosableProcess.runBlocking(commands.toArray(new String[0]));
 	}
 
-	public void copyOptJarsToLib(String jarNamePrefix) throws FileNotFoundException, IOException {
-		final Optional<Path> reporterJarOptional;
+	public void copyOptJarsToLib(String jarNamePrefix) throws IOException {
+		final Path jar = getOptJar(jarNamePrefix);
+		Files.copy(jar, lib.resolve(jar.getFileName()));
+	}
+
+	public void addPlugin(String jarNamePrefix) throws IOException {
```

Comment:
rename to `copyOptJarsToPlugins` for consistency; the current naming also makes it seem like you can add an arbitrary jar.